src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.cs

```diff
@@ -26,21 +27,21 @@ internal static IntPtr LoadLibraryByName(string libraryName, Assembly assembly,
             // First checks if a default dllImportSearchPathFlags was passed in, if so, use that value.
             // Otherwise checks if the assembly has the DefaultDllImportSearchPathsAttribute attribute.
             // If so, use that value.
-
-            if (!searchPath.HasValue)
+            bool userSpecifiedSearchFlags = searchPath.HasValue;
+            if (!userSpecifiedSearchFlags)
             {
-                searchPath = GetDllImportSearchPath(assembly);
+                searchPath = GetDllImportSearchPath(assembly, out userSpecifiedSearchFlags);
             }
-            return LoadLibraryByName(libraryName, assembly, searchPath.Value, throwOnError);
+            return LoadLibraryByName(libraryName, assembly, userSpecifiedSearchFlags, searchPath!.Value, throwOnError);
```

Comment:
> searchPath!.Value

Logic around null hasn't changed - filed https://github.com/dotnet/roslyn/issues/78288 for CS8629 being fired without the null-forgiving.